### DISCUSSION TITLE
Parallel solving returns unknown status

### DISCUSSION THREAD

**Discussion by nthe** [google/or-tools#2873] [CP-SAT questions] [answered]
Hi,
I'm using the CP-SAT solver for detailed employee scheduling on Windows 10, Python 3.9, and OR-Tools 9.1. 
My machine has 8 CPUs and 16 GB of RAM. 

The model is designed to schedule whole shifts which are made of different work tasks and breaks. The duration of tasks and breaks may vary and the work tasks should ideally satisfy forecasted staffing requirements. The agents have skills that are used for task type selection. As you can imagine, the model is quite complex, so in order to create a schedule for let's say a week, I have to solve schedules per day and post-process the results once all finished. 

Consider the following piece of code I'm using for sequential or parallel execution.
```py
data_chunks = chunk_data(data)
worker = partial(__optimize_domain, options)

if options.in_parallel:
    with futures.ProcessPoolExecutor() as executor:
        results = executor.map(worker, data_chunks)
    results = list(results)
else:
    results = list(map(worker, data_chunks))
```

When I run in **sequence**, all data chunks are processed with an `optimal` or `feasible` solution. 
But when I run it in **parallel**, some chunks finish and some lead to `unknown` status.

I also tried to force parallel workers to solve the same data chunk, but it led to the same result.
```py
data_chunks = [
    data_chunks[0], data_chunks[0], data_chunks[0]
]
```

Here's the solver response stats for a successful data chunk.
```txt
CpSolverResponse summary:
status: FEASIBLE
objective: 613272
best_bound: 0
booleans: 9112
conflicts: 0
branches: 15472
propagations: 1112066
integer_propagations: 1207045
restarts: 15472
lp_iterations: 0
walltime: 5.57321
usertime: 5.57321
deterministic_time: 0.440606
primal_integral: 1.43441
```

And here's the solver response stats for a failed one.
```txt
CpSolverResponse summary:
status: UNKNOWN
objective: 2511976
best_bound: 0
booleans: 9024
conflicts: 0
branches: 11662
propagations: 853015
integer_propagations: 926296
restarts: 11662
lp_iterations: 0
walltime: 5.49491
usertime: 5.49491
deterministic_time: 0.425744
primal_integral: 1.23221
```

I'm aware that the unknown status means that the search limit was reached, but I'm not aware of or am able to identify what might be the source of an issue or which limit or threshold can be relaxed (by looking at the https://github.com/google/or-tools/blob/stable/ortools/sat/sat_parameters.proto).

Any help or advice would be great. 🙏 

Here're exported models in text and binary forms for both chunks above. 
[exported-models.zip](https://github.com/google/or-tools/files/7408625/exported-models.zip)

**Comment by lperron** [comment, marked as answer]
can you print the log ?

parameters.log_search_progress = true

**Reply by nthe** [reply]
🤦‍♂️ I feel embarrassed. 

I've accidentally set `max_time_in_seconds` which was reached before the first solution. The solver managed to get to the first solution before that limit when executed in sequence. 

Sorry for wasting your time and thanks for the OR-Tools, it's an amazing tool. 👍 

Case closed.
